### PR TITLE
sickchill 2023.5.30 (new formula)

### PR DESCRIPTION
**Pull request by miigotu**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by miigotu** [comment]
I'm not sure if I'm allowed to add this here. I am perfectly content to maintain this in https://github.com/sickchill/homebrew-sickchill but I can't figure out how to habe arm64 built with github actions. Is there some documentation for this or a place I can discuss it with someone in real-time?

**Comment by nandahkrishna** [comment]
> I'm not sure if I'm allowed to add this here. I am perfectly content to maintain this in https://github.com/sickchill/homebrew-sickchill but I can't figure out how to habe arm64 built with github actions. Is there some documentation for this or a place I can discuss it with someone in real-time?

I think GitHub doesn't offer Apple Silicon machines yet, but they have support for self-hosted runners. You would have to look for a CI provider with Apple Silicon runners, but since your repository (https://github.com/sickchill/sickchill) is popular, it would be fine for us to add it to `homebrew-core` and we'd be able to provide bottles for Intel and Apple Silicon. Once it's added, you could even consider redirecting people to install directly from homebrew-core without having to install your tap.

File: Formula/sickchill.rb
Comment (nandahkrishna) [inline Formula/sickchill.rb:4]:
```suggestion
  desc "Media library mega tool"
```

File: Formula/sickchill.rb
Comment (nandahkrishna) [inline Formula/sickchill.rb:14]: This will be added automatically when the formula is ready to be merged.
```suggestion
```

File: Formula/sickchill.rb
Comment (nandahkrishna) [inline Formula/sickchill.rb:7]: It would be good to add the `license` as well.

File: Formula/sickchill.rb
Comment (nandahkrishna) [inline Formula/sickchill.rb:411]:
```suggestion
```

File: Formula/sickchill.rb
Comment (nandahkrishna) [inline Formula/sickchill.rb:425]: We don't add caveats unless they're _absolutely_ necessary and Homebrew-specific.
```suggestion
```

File: Formula/sickchill.rb
Comment (nandahkrishna) [inline Formula/sickchill.rb:429]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

**Review by nandahkrishna — COMMENTED** [review]
Some suggestions to improve the formula. When you've made these changes, please squash together all your commits into a single one with the message:
```
sickchill 2023.5.30 (new formula)
```
which is the same as the PR title.

File: Formula/sickchill.rb
Comment (miigotu) [reply Formula/sickchill.rb:14]: Yeah that was added by the homebrew-bot in my tap

File: Formula/sickchill.rb
Comment (miigotu) [reply Formula/sickchill.rb:7]: Yep, I added it on my tap, I will add it here as soon as that builds

File: Formula/sickchill.rb
Comment (miigotu) [reply Formula/sickchill.rb:416]: Maybe I can run it and kill the process somehow and check that var/"sickchill"/config.ini exists? I don't really know ruby, so how do I kill the process in the test?

**Comment by miigotu** [comment]
@nandahkrishna any idea what I'm doing wrong to get cryptography to build? I know there are prebuilt wheels for macos but building from source is tricky since rust.

File: Formula/sickchill.rb
Comment (nandahkrishna) [inline Formula/sickchill.rb:14]:
```suggestion
  # `pkg-config`, `rust`, and `openssl@3` are for cryptography.
  depends_on "pkg-config" => :build
  depends_on "rust" => :build
  depends_on "cffi"
  depends_on "openssl@3"
  depends_on "pycparser"
  depends_on "python@3.11"
  depends_on "pyyaml"
```
This should hopefully fix the `cryptography` build.

File: Formula/sickchill.rb
Comment (nandahkrishna) [inline Formula/sickchill.rb:58]:
```suggestion
```

File: Formula/sickchill.rb
Comment (nandahkrishna) [inline Formula/sickchill.rb:228]:
```suggestion
```

File: Formula/sickchill.rb
Comment (nandahkrishna) [inline Formula/sickchill.rb:318]:
```suggestion
```

File: Formula/sickchill.rb
Comment (nandahkrishna) [reply Formula/sickchill.rb:416]:
```suggestion
  test do
    port = free_port.to_s
    fork { exec bin/"sickchill", "--port", port, "--datadir", testpath, "--nolaunch" }
    sleep 5
    assert_predicate testpath/"config.ini", :exist?
  end
```
Something like this maybe?

**Comment by miigotu** [comment]
I think there's a file I need to create to avoid the homebrew-pypi-poet from re-adding those resource stanzas back from my pyproject.toml on the next update?

**Comment by miigotu** [comment]
@nandahkrishna I believe this is ready, everything is passing and I don't think anything was missed this time! Let me know what else I can do to improve it.

Also, can I create a tap_migrations.json on my projects tap to redirect to core once this is merged? Would that successfully deprecate our tap without breaking our users installs? I have not idea how many of our 50k or so users have tried my tap, and that way I don't have to worry about redirecting all of the early adopters of my tap. I can then simply make notes of the migration in our documentation. Thanks so much for your help, I knew literally zero ruby before like a week ago 😂

**Review by nandahkrishna — APPROVED** [review]
Thanks for your contribution! I'll leave this open for a while just in case others have feedback. And yes once this is merged, you can add a `tap_migrations.json` to your Tap to migrate your users seamlessly.

**Comment by miigotu** [comment]
@carlocab do you mind looking this over if you have time? I'm not sure who else to get a 2nd review from, I just saw you do quite a bit of reviews =P Thanks!

**Comment by nandahkrishna** [comment]
Should be okay to merge, just slipped my mind. Thanks again!

**Pull request merged**
